components/central-application-gateway/internal/proxy/proxy.go

```diff
@@ -77,10 +82,10 @@ func (p *proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	cacheEntry.Proxy.ServeHTTP(w, newRequest)
 }
 
-func (p *proxy) extractPath(u *url.URL) (model.APIIdentifier, string, *url.URL, apperrors.AppError) {
+func (p *proxy) extractPath(u *url.URL) (model.APIIdentifier, *url.URL, *url.URL, apperrors.AppError) {
```

Comment:
These return types are getting quite hard to guess what is what ;-; It's not a part of this task so I don't think you should fix it here, just noticed stuff :d